README.md

```diff
@@ -17,7 +17,113 @@ TerseTS can be compiled and cross-compiled from source:
    - Microsoft Windows: `zig build -Dtarget=x86_64-windows -Doptimize=ReleaseFast`
 
 # Usage
-TODO
+
+To use TerseTS in a C project, follow these steps:
+
+1. **Installation**:
+   - Ensure TerseTS is installed and compiled as per the installation instructions.
+
+2. **Linking**:
+   - For Windows: Link the `tersets.dll` to your project.
+   - For Linux: Link the `libtersets.so` to your project.
+   - Include the header file `tersets/bindings/c/tersets.h` in your project.
+
+3. **Include the Header**:
+   - Add `#include "tersets.h"` in your source files where you intend to use TerseTS functions. 
+
+4. **Compression and Decompression Interface**:
+   - Use the provided interface to compress and decompress time series data.
+
+### Example
+
+```c
+#include "tersets.h"
+#include <stdio.h>
+
+int main() {
+    // Example uncompressed data
+    double data[] = {1.0, 2.0, 3.0, 4.0, 5.0};
+    struct UncompressedValues uncompressed_values = {data, 5};
+
```

Comment:
Maybe print the length of the uncompressed data as well so the user can see the length of uncompressed, compressed, and decompressed data.

Reply:
Fixed. Now all example print the length of all variables.